code/modules/plumbing/ducts.dm

```diff
@@ -359,12 +363,18 @@ All the important duct code:
 		if(!D.anchored)
 			add(1)
 			qdel(D)
+			return
+
+		to_chat(user, span_warning("The duct must be unanchored before it can be picked up."))
+		return
```

Comment:
swap these checks, check if D is anchored and return if it is, then do the rest of the code unindented.

aka early return